src/routes/v2/authenticatedSites/repoManagement.ts

```diff
@@ -0,0 +1,71 @@
+import autoBind from "auto-bind"
+import express from "express"
+
+import { BadRequestError } from "@errors/BadRequestError"
+import { ForbiddenError } from "@errors/ForbiddenError"
+
+import { AuthorizationMiddleware } from "@middleware/authorization"
+import { attachWriteRouteHandlerWrapper } from "@middleware/routeHandler"
+
+import UserWithSiteSessionData from "@root/classes/UserWithSiteSessionData"
+import GitFileSystemError from "@root/errors/GitFileSystemError"
+import { attachSiteHandler } from "@root/middleware"
+import type { RequestHandler } from "@root/types"
+import RepoManagementService from "@services/admin/RepoManagementService"
+
+interface RepoManagementRouterProps {
+  repoManagementService: RepoManagementService
+  authorizationMiddleware: AuthorizationMiddleware
+}
+
+export class RepoManagementRouter {
+  private readonly repoManagementService
+
+  private readonly authorizationMiddleware
+
+  constructor({
+    repoManagementService,
+    authorizationMiddleware,
+  }: RepoManagementRouterProps) {
+    this.repoManagementService = repoManagementService
+    this.authorizationMiddleware = authorizationMiddleware
+    autoBind(this)
+  }
+
+  resetRepo: RequestHandler<
+    never,
+    void | { message: string },
+    { branchName: string; commitSha: string },
+    unknown,
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { userWithSiteSessionData } = res.locals
+    const { branchName, commitSha } = req.body
+
+    return this.repoManagementService
+      .resetRepo(userWithSiteSessionData, branchName, commitSha)
+      .map(() => res.status(200).send())
+      .mapErr((error) => {
+        if (error instanceof BadRequestError) {
+          return res.status(400).json({ message: error.message })
+        }
+        if (error instanceof ForbiddenError) {
+          return res.status(403).json({ message: error.message })
+        }
+        if (error instanceof GitFileSystemError) {
+          return res.status(500).json({ message: error.message })
+        }
+        return res.status(502).json({ message: error.message })
+      })
+  }
+
+  getRouter() {
+    const router = express.Router({ mergeParams: true })
+    router.use(attachSiteHandler)
+    router.use(this.authorizationMiddleware.verifySiteMember)
+
+    router.post("/resetRepo", attachWriteRouteHandlerWrapper(this.resetRepo))
```

Comment:
This is only meant for E2E repo right? If so, we should nest within /e2e

If this is meant as a generic endpoint to reset repos to a specific commit then this would be fine

Reply:
Yup it is intentional, I'm leaving the option open in the future in case we might ever need this endpoint for some other reason via the CMS frontend. For now however, the service also checks that it is within the e2e test repos, so the functionality is very limited.

Reply:
sgtm